build-tools/packages/build-cli/knip.config.ts

```diff
@@ -0,0 +1,27 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// Enable TypeScript type-checking for this file.
+// See https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check
+// @ts-check
+
+// Import the shared config from the root of the repo.
+import sharedConfig from "../../../knip.base.ts";
+
+export default {
```

Comment:
Add a doc comment with a type directive here so we can get type checking on the config object. Something like this:

```
/**
 * @type {import("knip").Config}
 */
```

Reply:
Oh right it's typescript - just `import type` then.

Reply:
And you can remove the ts-check-related comments.